src/content/extension.js

```diff
@@ -1,39 +1,24 @@
 import GitHub from "./github";
 import LeetCode from "./leetCode";
 import StackOverflow from "./stackOverflow";
+
 import Storage from "./storage";
 
-const GITHUB_URL = "https://github.com";
-const STACKOVERFLOW_URL = "https://stackoverflow.com";
-const STACKOVERFLOW_VALID_PATHNAMES = /(^\/questions|\/posts\/\d+\/edit|^\/review)/u;
-const LEETCODE_URL = "https://leetcode.com";
-const LEETCODE_VALID_PATHNAMES = /(^\/problems)/u;
+const extensions = [GitHub, StackOverflow, LeetCode];
 
-export default class Extension {
+export default class {
   constructor() {
-    this._buttons = null;
     this._storage = new Storage();
   }
 
   async init() {
     await this._storage.init();
 
-    if (window.location.origin === GITHUB_URL) {
-      this._buttons = new GitHub(this._storage);
-    }
-
-    if (
-      window.location.origin === STACKOVERFLOW_URL &&
-      STACKOVERFLOW_VALID_PATHNAMES.test(window.location.pathname)
-    ) {
-      this._buttons = new StackOverflow(this._storage);
-    }
-
-    if (
-      window.location.origin === LEETCODE_URL &&
-      LEETCODE_VALID_PATHNAMES.test(window.location.pathname)
-    ) {
-      this._buttons = new LeetCode(this._storage);
+    for (const Extension of extensions) {
+      if (typeof Extension.test === "function" && Extension.test()) {
```

Comment:
It doesn't look like any of the extensions are missing `#test()`. If that's true, I'd prefer not to have this guard so that it actually throws if one of the Extensions doesn't have this static method rather than silently not work.

Reply:
agreed. removed.